packages/@aws-cdk/aws-stepfunctions-tasks/lib/emr/emr-create-cluster.ts

```diff
@@ -319,6 +319,16 @@ export class EmrCreateCluster extends sfn.TaskStateBase {
       );
     }
 
+    // Allow the StateMachine to tag clusters if tags are defined
+    if (this.props.tags) {
+      policyStatements.push(
+        new iam.PolicyStatement({
+          actions: ['elasticmapreduce:AddTags'],
+          resources: ['*'],
```

Comment:
Is it possible to scope this down further?

Reply:
As per https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonelasticmapreduce.html, AddTags action supports four kinds of EMR resources. Clusters are the only ones this state needs to be able to tag. The resource could be limited to all cluster resources (wildcard still required on the clusterid part of the ARN).

As for conditions, the action seems to support the usual RequestTag style condition keys. We could, in theory, add a condition to this statement that limits the tag keys and their values to the values specified in the tags prop. However, this gets complex pretty quickly since e.g. tag values can be dynamic jsonpath references to fields in the state object and tag keys can (maybe) even be CDK tokens. Constructing RequestTag conditions that support all these edge cases tricky very fast. Plus, if the user is allowed to use this CDK construct and create a role for the state machine, they could anyway specify any tags they wish in the props and hence the conditions generated based on them would allow any tags chosen by the user.  

A condition I would maybe like to see here is something that would allow tagging to be limited to happen during RunJobFlow call only. EC2 has ec2:CreateAction condition key which can be used to do that. With this kind of limitation you would not be able to go around tagging existing clusters if you happen to have access to execute a state machine that accepts tag keys / values from the state object. However, EMR does not seem to support such condition so that cannot be done.

Anyway, I'll change the resource to only allow tagging on clusters and leave out the additional conditions due to complications explained above. Thanks for the comment!